src/components/article-preview.js

```diff
@@ -6,15 +6,27 @@ import styles from './article-preview.module.css'
 
 export default ({ article }) => (
   <div className={styles.preview}>
-    <Img alt="" fluid={article.heroImage.fluid} />
+    { 
+      article.heroImage &&
+      article.heroImage.fluid &&
+      <Img alt="" fluid={article.heroImage.fluid} />
```

Comment:
Gatsby supports https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining so lets use it :)